packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -213,7 +460,7 @@ class Order extends Entity {
   static definition = new ModelDefinition('Order')
     .addProperty('id', {type: 'number', id: true})
     .addProperty('description', {type: 'string', required: true})
-    .addProperty('customerId', {type: 'number', required: true})
+    .addProperty('customerId', {type: 'number', required: false})
```

Comment:
Interesting. How can an order exist when there is no customer placing it? Is it a limitation of the current implementation of has-many-through relation that it's not possible to enforce "required" constraint on foreign keys?

Maybe I am misunderstanding this change. I see that you are adding `CustomerOrderLink` model to establish N-N relation between customers and orders. IMO, `Order` should not have `customerId` property in that scenario.

I think it's not a good idea to reuse customer-order relation for has-many-through and we should introduce new models & relations for that.

Reply:
👍 make sense. Added a new model `CartItem`. A `Customer` has many `CartItem`s through `CustomerCartItemLink`.